x11-packages/qt-creator/build.sh

```diff
@@ -0,0 +1,34 @@
+TERMUX_PKG_HOMEPAGE=https://www.qt.io/
+TERMUX_PKG_DESCRIPTION="Integrated Development Environment for Qt"
+TERMUX_PKG_LICENSE="GPL-3.0-only"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION="18.0.0"
+TERMUX_PKG_SRCURL="https://download.qt.io/official_releases/qtcreator/${TERMUX_PKG_VERSION%.*}/${TERMUX_PKG_VERSION}/qt-creator-opensource-src-${TERMUX_PKG_VERSION}.tar.gz"
+TERMUX_PKG_SHA256=6c8bcbd2f33aa6851f265ab22fc8bee7fc4cfcb3cc1d0db73978c7afe9272eb3
+TERMUX_PKG_AUTO_UPDATE=true
+TERMUX_PKG_DEPENDS="clang, glib, libandroid-execinfo, libarchive, libelf, libllvm, libsecret, opengl, python, qt6-qtbase, qt6-qtcharts, qt6-qtdeclarative, qt6-qttools, qt6-qtsvg, libyaml-cpp, zstd"
```

Comment:
If QtCreator 18 *requires* `-DLLVM_ENABLE_RTTI=ON` it should probably have a version caveat on the dependency.
```suggestion
TERMUX_PKG_DEPENDS="clang, glib, libandroid-execinfo, libarchive, libelf, libllvm (>> 21.1.5), libsecret, opengl, python, qt6-qtbase, qt6-qtcharts, qt6-qtdeclarative, qt6-qttools, qt6-qtsvg, libyaml-cpp, zstd"
```

Reply:
The thing is, if anyone installs this and _doesn't_ have `libllvm` version 21.1.5-1  (or newer) then it means they performed a **partial upgrade** and there is no other way they can get that state.

There are so many other ways that a **partial upgrade** can break Termux that it's impossible to fix all of them by marking `(>> X.Y.Z)` all over the place. That is just how all rolling-release distros work.

Unlike the situation with, for example, `qt6-qtwayland` and some other packages that have versioned dependencies in Termux, the error occurs at **runtime**, not at **install-time**, therefore the only hypothetical reason to have a versioned dependency would be because of a partial upgrade.

https://github.com/termux/termux-packages/blob/190afe4641dc604e475275fb15f21e08a0201d1c/x11-packages/qt6-qtbase/build.sh#L14

I prefer this global solution to partial upgrades for all people using `pkg install` on anything:

- https://github.com/termux/termux-tools/pull/172

if discussing potential measures to prevent partial upgrades, I prefer to focus on that PR rather than on individual Termux packages, because that PR can help with the problem globally for all packages simultaneously.

Reply:
- Mmmh I see, so a standard "termux/termux-tools#172 would have prevented this" situation.